lib/bap_llvm/llvm_loader_stubs.c

```diff
@@ -18,14 +18,17 @@ static void loader_fail(const struct bap_llvm_loader *loader, int n) {
     failn(n);
 }
 
-value bap_llvm_load_stub(value arg) {
-    CAMLparam1(arg);
+value bap_llvm_load_stub(value arg, value pdb_path) {
+    CAMLparam2(arg, pdb_path);
     CAMLlocal1(result);
     const struct caml_ba_array* array = Caml_ba_array_val(arg);
     if ((!array->dim[0]) || (array->num_dims != 1))
         failn(1);
+    const char * pdb = NULL;
+    if (caml_string_length(pdb_path))
+        pdb = String_val(pdb_path);
```

Comment:
the `pdb` pointer will be invalidated on the next call to OCaml runtime. So if `bap_llvm_loader_create` uses any OCaml callbacks (or will use in future) it will segfault. It is safer to strdup it (and then free at the end of the function). When freeing make sure that the `pdb` pointer is not stored somewhere in LLVM (on the first glance it shouldn't, but it is better to be sure, we are here to detect UAF not to create them).